### ISSUE TITLE
Edit option missing for company managers and site admin

### ISSUE THREAD

**Issue by sarathp06** [iomad/iomad#1710]
I have upgraded my IOMAD_310 site to IOMAD_400_STABLE. While checking as a company manager, I could not see the edit option from mysite/blocks/iomad_company_admin/editusers.php. In the previous version, managers and even site admin were able to edit, re-send user details, delete and suspend. But now those options are missing. Please find the screenshot below:

![edituser](https://user-images.githubusercontent.com/54504917/187059630-d79f11cb-ff98-4d53-a1e5-cdb301b97e7a.jpg)
.
Do I have to enable it from somewhere?

Thanks in advance.

**Comment by turf212** [comment]
Much of the editing functionality has been hidden until you click on the switch for Edit mode.  You can now (as of 4.0) also promote/demote users and change their department assignations from this page too.

**Comment by sarathp06** [comment]
Hi Derick,

When Edit mode is on, it's showing "Coding error detected, it must be fixed by a programmer: Options for select element must contain an option for the specified value
More information about this error".
Please find the screenshot below:
![editusers_turn_editon](https://user-images.githubusercontent.com/54504917/188132338-76ca13cc-1e8b-4b1d-a358-f7bcd9e19d4c.jpg)

When I edit mode is off, it will show the users list.

**Comment by turf212** [comment]
Can you please turn on full debug (Site administration -> Development -> debugging and change type to full developer debug, ensure display error messages is ticked and save)

**Comment by sarathp06** [comment]
Hi Derick,

Sure. Please find the below screenshot:
![iomad_edit_users](https://user-images.githubusercontent.com/54504917/188155896-723b2887-d1bd-4253-b771-a59e759c0cf4.jpg)

**Comment by turf212** [comment]
The selector isn't being passed any roles which can be added.  The table class builds this array and should always have an entry for ['0' => get_string('user', 'block_iomad_company_admin')] with the rest being built from permissions.

Can you check what $this->usertypeselect has as values in that table class just before line 137?

**Comment by sarathp06** [comment]
This is the code before line 137:
![code_iomad](https://user-images.githubusercontent.com/54504917/188159823-dd3b91f9-a893-427b-addc-c14c600f53aa.jpg)

I did a var_dump($usertypeselect);
die;
And the result is:
![result](https://user-images.githubusercontent.com/54504917/188160108-23db513a-02a4-4ba1-96ef-62940ac24009.jpg)

**Comment by turf212** [comment]
Which is exactly what I would expect.

What is $currentvalue set to?

**Comment by sarathp06** [comment]
The current value is int(11).
![current_Val](https://user-images.githubusercontent.com/54504917/188173299-7f9b0ada-3d8e-4c13-9dc0-388573962bfe.JPG)

**Comment by turf212** [comment]
It's a data issue.  The user is type 11 (Educator only) but this isn't in the list of available selections.  Likely this is due to you changing the setting for the Local IOMAD plugin -> Enrol managers as non students to unchecked, promoting someone as an educator and then re-checking this option.  If you untick this option in the plugin settings you will be able to sort the underlying issue (either leave it as is as it's more flexible) or demote all of the educators and then re-select the option.

**Comment by turf212** [comment]
Oh - another thought - it could also be that you have a company manager assigned to a department which is not the top level department for the company (parent = 0 in mdl_department) which is not within the expected parameters.

**Comment by sarathp06** [comment]
Ok. However, I haven't changed any settings. Now I installed a fresh IOMAD_400_STABLE, created a company, then created a user, and assigned the role "company manager" from the IOMAD admin block. Then I logged in as the company manager and while the Edit mode is on, I went to edit users and there I am getting the above error. I am also getting the same errors on that page when I logged in as site administrator. 
As you mentioned, If I uncheck "Enrol managers as non students" the above errors are gone for both site admin and company managers.

I have created only one company(Company A) and therefore while creating a user, the department is selected as "Company A". I haven't created any separate Departments.
